src/Cli/dotnet/Commands/Test/VSTest/TestCommand.cs

```diff
@@ -298,11 +298,16 @@ internal static int RunArtifactPostProcessingIfNeeded(string testSessionCorrelat
 
     private static bool ContainsBuiltTestSources(string[] args)
     {
-        foreach (string arg in args)
+        for (int i = 0; i < args.Length; i++)
```

Comment:
What if I run `dotnet test --filter "Category=.dll"`?
Is this going to incorrectly return true?